### PR TITLE
feat(api): GraphQL Custom Request Headers

### PR DESCRIPTION
**Pull request by Equartey**
*Issue #, if available:*
https://github.com/aws-amplify/amplify-flutter/issues/1217

*Description of changes:*
Allows devs to pass custom headers for GraphQL http requests. 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: packages/amplify_core/lib/src/types/api/graphql/graphql_request.dart [hunk 1]
```diff
@@ -22,6 +22,9 @@ class GraphQLRequest<T> {
   /// Only required if your backend has multiple GraphQL endpoints in the amplifyconfiguration.dart file. This parameter is then needed to specify which one to use for this request.
   final String? apiName;
 
+  /// A map of Strings to dynamically use for custom headers in the http request.
+  final Map<String, String>? customHeaders;
```
Comment (dnys1) [inline packages/amplify_core/lib/src/types/api/graphql/graphql_request.dart:26]: What other kind of headers are there? ;-) I would just call em `headers`

**Review by dnys1 — APPROVED** [review]
LGTM. Will want some unit tests verifying their presence once auth providers enter the party.

File: packages/amplify_core/lib/src/types/api/graphql/graphql_request.dart (hunk 1 above)
Comment (Equartey) [reply packages/amplify_core/lib/src/types/api/graphql/graphql_request.dart]: Good point, updated!

**Review by Jordan-Nelson — APPROVED** [review]
lgtm

**Pull request merged**